### PR TITLE
Add beginner content

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Core idea is to add a series of beginner content with some theory and practice with nanoFramework pointing on the rest of the documentation and samples

## Motivation and Context

- [x] Gpio
- [x] Spi
- [x] I2c
- [x] DAC
- [x] ADC
- [x] UART
- [x] Add a summary documentation and links to this new documentation in various places

## Types of changes

- [ ] Improvement (correction, content improvement, typo fix, formatting)
- [x] New Article (new document for docs website)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [x] My doc follows the code style of this project.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -722,6 +722,16 @@ steps:
       } else {
         Write-Host "The folder does not exist most likely because the devices hasn't been built."
       }
```
Comment (josesimoes) [inline azure-pipelines.yml:724]: These changes here in the YAML must have been unwillingly added.

File: content/getting-started-guides/gpio-explained.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+- Input/Output: GPIO pins can either receive information (input) from external devices or send out information (output) to control external components. For instance, an input GPIO pin can detect whether a button is pressed or not, while an output GPIO pin can send a signal to turn on an LED or control a motor.
+
+In embedded systems, GPIO pins are versatile. They're used for a wide range of tasks, including reading sensors, interfacing with buttons or switches, controlling LEDs, motors, and communicating with other devices like displays or communication modules.
+
+Programmers can write code to configure these GPIO pins based on the specific requirements of their project. By manipulating these pins through code, developers can create complex functions and interactions within an embedded system. Those GPIO pins are used as a base for other higher level protocols like SPI, I2C, Serial and more!.
+
+Overall, GPIO provides a flexible way to interact with the physical world by allowing a microcontroller to interface with various external devices and components in an embedded system.
+
+## A bit of electronic
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:18]:
```suggestion
## A bit of electronics theory
```

File: content/getting-started-guides/gpio-explained.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+`V = R x I`
+
+Where:
+
+- `V` = voltage in Volt (V)
+- `R` = resistor in Ohms
+- `I` = intensity in Amperes (A)
+
+Seems like useless but it's what really matter when you connect elements together.
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:30]:
```suggestion
Seems like useless but it's what really matter when you connect the elements together.
```

File: content/getting-started-guides/gpio-explained.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+> [!IMPORTANT]
+> Playing with electricity is sensitive for MCU, the maximum voltage is a real maximum voltage, go over and you'll damage the pin and possibly the MCU
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:33]:
```suggestion
> Playing with electricity is sensitive for MCU, the maximum voltage is a real maximum voltage, go over and you'll damage the pin and possibly the MCU
```
```suggestion
> Playing with electricity is sensitive for MCU, the maximum voltage is a real maximum voltage, go over and you'll damage the pin and possibly the MCU.
```

File: content/getting-started-guides/gpio-explained.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+> Check in the documentation what is the maximum level of the pin
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:34]:
```suggestion
> Check in the documentation what is the maximum voltage the pins are tolerant to.
```

File: content/getting-started-guides/gpio-explained.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+> ESP32, STM32 and most MCU are 3.3V, some can be 1.8V and some even tolerant to higher voltages like 5V, meaning that they are 3.3V but you can apply in their input 5V (sometimes more) and they'll be perfectly fine!
+
+Also, few terms we will use:
+
+- Ground = 0V = 0 = Low
+- VCC = +Power, can be +5V, more often in MCU world +3.3V = 1 = High
+
+## Controlling a led with GPIO a bit more theory
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:42]:
```suggestion
## Controlling a LED with GPIO a bit more theory
```

File: content/getting-started-guides/gpio-explained.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+The basic hello world in the embedded space is blinking a led. For this, we need a bit of electronic knowledge and some hardware:
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:44]:
```suggestion
The basic hello world in the embedded space is blinking a LED. For this, we need a bit of electronic knowledge and some hardware:
```

File: content/getting-started-guides/gpio-explained.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+- 1 GPIO pin available as output on a MCU
+- 1 Led
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:47]:
```suggestion
- 1 LED
```

Comment (josesimoes) [reply content/getting-started-guides/gpio-explained.md:47]: There are other instances of _led_ in low caps in the text below.

File: content/getting-started-guides/gpio-explained.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+- 1 Resistor
+
+And here is how look like the electronic schema, we will take the assumption our MCU provides 3.3V:
+
+![R and a led to ground](../../images/getting-started/gpio-pin-schema.png)
+
+Each electronic component has a limit either in terms ov voltage you can apply/it can handle, either in terms of intensity or both. That is the case for any led. It does vary a bit depending on the led and its color but overall, 1 led is approximately 2 V and can handle a current of 0,02 A maximum.
+
+Applying the Ohms law: `R = (3,3 - 2)/0,02 = 65 Ohms`
```
Comment (josesimoes) [inline content/getting-started-guides/gpio-explained.md:56]:
```suggestion
Applying Ohm's law: `R = (3,3 - 2)/0,02 = 65 Ohms`
```

**Review by josesimoes — COMMENTED** [review]
Nice content!! Just some typos and upper/lower case.

File: azure-pipelines.yml (hunk 1 above)
Comment (Ellerbach) [reply azure-pipelines.yml:724]: Yes, I'm going to have another PR, just wanted to test this.

File: content/getting-started-guides/gpio-explained.md [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+
+## A bit of theory
+
+GPIO stands for "General Purpose Input/Output." In simple terms, [GPIO](https://en.wikipedia.org/wiki/General-purpose_input/output) refers to a set of pins on a microcontroller (MCU) or a computer that can be programmed to either input or output signals. These signals can be used to communicate with and control other devices or components in an embedded system.
+
+- General Purpose: These pins can be configured by the user or programmer for various tasks. They're not dedicated to one specific function but can be adapted for different purposes based on the needs of the system.
+- Input/Output: GPIO pins can either receive information (input) from external devices or send out information (output) to control external components. For instance, an input GPIO pin can detect whether a button is pressed or not, while an output GPIO pin can send a signal to turn on an LED or control a motor.
```
Comment (torbacz) [inline content/getting-started-guides/gpio-explained.md:10]: If it's for beginners I would place something like:
"NOTE! Gpio pins are able to deliver low current. Connecting high demand devices may lead to destroy them"

**Comment by CoryCharlton** [comment]
Not sure if it's just me or the Github preview but most of the images are extremely hard to read (transparent background?):

image

**Comment by Ellerbach** [comment]
> Not sure if it's just me or the Github preview but most of the images are extremely hard to read (transparent background?):

They are using transparent background as they are png. The main intent is to be on https://docs.nanoframewrok.net. Where the default style is with a white background.

File: content/getting-started-guides/adc-explained.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
+# All what you've always wanted to know about ADC
+
+This beginner document will explain what is ADC and how to properly use it with concrete examples using .NET nanoFramework. You're in a perfect place if you want to have analogic work done and reading analogic values from a pin.
+
+## A bit of theory
+
+An Analog-to-Digital Converter (ADC) in the realm of microcontrollers (MCU) is a fundamental component that translates analog signals, like temperature or voltage, into digital data that the MCU can understand and process. It enables MCU to interact with and interpret real-world analog signals, converting them into a digital format for analysis, storage, or further manipulation within the digital realm of the MCU. ADC is is opposite of [Digital-to-Analog Converter (DAC)](./dac-explained.md).
```
[the remark this answers is not in the document]
Comment (Ellerbach) [reply content/getting-started-guides/adc-explained.md:7]: yes, it is indeed :-) reason I used those as examples. Anything unclear with this?

[the remark this answers is not in the document]
Comment (torbacz) [reply content/getting-started-guides/adc-explained.md:7]: Well, from technical point of view it's always represented as voltage, then you need to convert it into data. If I doesn't know how ADC works, I might assume after reading from ADC chanel I would get temperature value.

[the remark this answers is not in the document]
Comment (Ellerbach) [reply content/getting-started-guides/adc-explained.md:7]: Haaaaa, totally got your point. Let me for sure clarify! Thanks again for the feedback!

**Pull request merged**